src/main/java/org/apache/commons/io/FilenameUtils.java

```diff
@@ -1490,4 +1496,163 @@ public static boolean wildcardMatch(final String fileName, final String wildcard
         return list.toArray( new String[ list.size() ] );
     }
 
+    /**
+     * Checks whether a given string is a valid host name according to
+     * RFC 3986.
+     *
+     * <p>Accepted are IP addresses (v4 and v6) as well as what the
+     * RFC calls a "reg-name". Percent encoded names don't seem to be
+     * valid names in UNC paths.</p>
+     *
+     * @see "https://tools.ietf.org/html/rfc3986#section-3.2.2"
+     * @param name the hostname to validate
+     * @return true if the given name is a valid host name
+     */
+    private static boolean isValidHostName(String name) {
+        return isIPv4Address(name) || isIPv6Address(name) || isRFC3986HostName(name);
+    }
```

Comment:
isIPv4Address can return false for input that looks like an IP address but fails a subsequent check, e.g. 999.9.9.9.
In which case it will be subject to two further checks (which must fail if the code is correct).